### ISSUE TITLE
Ban multiple awaits

### ISSUE THREAD

**Issue by jpike88** [eslint/eslint#13918]
**Please describe what the rule should do:**

Ban chained await calls, without any separating brackets

**What new ECMAScript feature does this rule relate to?**

async await

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[x] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
const x = await await promiseFunc();
```

**Why should this rule be included in ESLint (instead of a plugin)?**

It's such an easy mistake to make, and I believe is an anti pattern. Should at least be separated by brackets or something.

**Are you willing to submit a pull request to implement this rule?**

No

**Comment by anikethsaha** [comment]
Thanks for the issue. 

This looks like a stylistic enhancement. 
As per our policy, no new stylistic enhancement or new stylistic rules are being accepted as of now. 

You can create a custom plugin for this.

**Comment by ljharb** [comment]
Double-awaiting creates additional microtasks, so it can actually cause bugs or performance problems - it's not purely stylistic.

**Comment by mdjermanovic** [comment]
> **What new ECMAScript feature does this rule relate to?**
> 
> async await

This doesn't qualify per our [rules policies](https://eslint.org/blog/2020/05/changes-to-rules-policies), since it isn't a feature added within the last 12 months, and it's also easy to configure [no-restricted-syntax](https://eslint.org/docs/rules/no-restricted-syntax) to catch the same error:

```js
/*eslint "no-restricted-syntax": ["error", "AwaitExpression > AwaitExpression.argument" ]*/

async () => {
  const x = await await promiseFunc(); // error
}
```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgXCJuby1yZXN0cmljdGVkLXN5bnRheFwiOiBbXCJlcnJvclwiLCBcIkF3YWl0RXhwcmVzc2lvbiA+IEF3YWl0RXhwcmVzc2lvbi5hcmd1bWVudFwiIF0qL1xuXG5hc3luYyAoKSA9PiB7XG4gIGNvbnN0IHggPSBhd2FpdCBhd2FpdCBwcm9taXNlRnVuYygpOyAvLyBlcnJvclxufVxuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo4LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

> It's such an easy mistake to make, and I believe is an anti pattern.

What kind of mistakes can lead to having a double-await in the code, is it a misunderstanding of how the promise resolution works or intent to manipulate the execution order?

**Comment by jpike88** [comment]
> What kind of mistakes can lead to having a double-await in the code, is it a misunderstanding of how the promise resolution works or intent to manipulate the execution order?

Accidentally typing it (or during cut/pasting), we actually never intend to write double await syntax in any of our code. The real issue is that when we do write it by accident, there was no obvious way for a linter to pick it up. The bugs that result can be bizarre and time-consuming when trying to resolve it.

I'm sure there is the weird/rare edge case where writing ```await await``` makes sense, but for our code I'd want to enforce a rule to prevent it ever happening.

Thanks for the above suggestion, I can just roll with that... feel free to close this if you consider it resolved

**Comment by mdjermanovic** [comment]
> Thanks for the above suggestion, I can just roll with that... feel free to close this if you consider it resolved

Glad it works! I'm closing this issue as there's a solution with `no-restricted-syntax`.
